### PR TITLE
Fix commitMessage and pass CMSContent through templates and routes

### PR DESCRIPTION
**Pull request by revgum**
See issue #7

### CODE REVIEW

File: core/utils/path.ts [hunk 1]
```diff
@@ -39,9 +34,9 @@ export function getPath(...parts: string[]) {
 
 const staticUrl = new URL(import.meta.resolve("../../static/"));
 
-export function asset(url = "") {
+export function asset(ctx: Context, url = "") {
   if (staticUrl.protocol === "file:") {
-    return posix.join(basePath, url);
+    return posix.join(ctx.var.options.basePath, url);
```
Comment (revgum) [inline core/utils/path.ts:39]: Passing context through the various places that `asset` or `getPath` are called eventually boils down to having the ability to use the `basePath` that the CMS was initialized with.

**Comment by oscarotero** [comment]
Thanks for this! Sorry for the commitMessage bug, I should test it before releasing.

Regarding to basePath issue, I like your approach but would rather to narrow the variable type to `CMSContent` instead of the entire context.

For example, instead of this:

```js
await collectionList({
  context: c,
  collection,
  version: await versioning?.current(),
}),
```

We can change it to this:

```js
await collectionList({
  options: c.var.options,
  collection,
  version: await versioning?.current(),
}),
```

This makes the templates more independent of Hono, reducing the internal dependency.

It could even be narrowed to:

```js
await collectionList({
  basePath: c.var.options.basePath,
  collection,
  version: await versioning?.current(),
}),
```
But passing the entire options object allows to include more values in the future without changing much code.

**Comment by revgum** [comment]
Seems reasonable! The PR is updated!

**Comment by oscarotero** [comment]
Thank you!

**Pull request merged**